src/test/java/qna/domain/AnswerRepositoryTest.java

```diff
@@ -3,48 +3,60 @@
 import static org.assertj.core.api.Assertions.assertThat;
 import static qna.domain.AnswerTest.A1;
 import static qna.domain.AnswerTest.A2;
-import static qna.domain.AnswerTest.A3;
 import static qna.domain.QuestionTest.Q1;
+import static qna.domain.UserTest.JAVAJIGI;
+import static qna.domain.UserTest.SANJIGI;
 
 import java.util.List;
 import java.util.Optional;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
-import org.mockito.Answers;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.TestConstructor;
 
+@TestConstructor(autowireMode = TestConstructor.AutowireMode.ALL)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
 @DataJpaTest
 class AnswerRepositoryTest {
 
-    @Autowired
-    private QuestionRepository questionRepository;
+    private final QuestionRepository questionRepository;
 
-    @Autowired
-    private AnswerRepository answerRepository;
+    private final AnswerRepository answerRepository;
 
+    private final UserRepository userRepository;
+
+    public AnswerRepositoryTest(QuestionRepository questionRepository, AnswerRepository answerRepository,
+                                UserRepository userRepository) {
+        this.questionRepository = questionRepository;
+        this.answerRepository = answerRepository;
+        this.userRepository = userRepository;
+    }
 
     @DisplayName("questionId와 일치하고 삭제가 안된 답들을 조회한다.")
     @Test
     void findByQuestionIdAndDeletedFalse() {
-        questionRepository.save(Q1);
+        userRepository.save(JAVAJIGI);
```

Comment:
```suggestion
        final val newJavajigi = userRepository.save(JAVAJIGI);
```

왜 이렇게 다시 newJavajigi 변수로 assign 할까요?

Reply:
save 메서드 내에서는 엔티티의 상태에 따라 `merge`(준영속 상태일 수 있기 때문)혹은 `persist`로 분기 처리됩니다. 따라서 실제 리턴되는 객체의 주솟값이 달라지게 되어 메서드 호출 시에는 받아서 할당하는 것이 더 편리합니다.